File: bindings/ruby/testsuite/function-call.rb
#!/usr/bin/ruby

require 'test/unit'
require 'storage'


class TestFunctionCall < Test::Unit::TestCase

  def test_function_call

    environment = Storage::Environment.new(true, Storage::ProbeMode_NONE, Storage::TargetMode_DIRECT)
    storage = Storage::Storage.new(environment)

    assert_equal(storage.default_mount_by(), Storage::MountByType_UUID)
    assert_raises(ArgumentError) { storage.default_mount_by("extra parameter") }
    assert_raises(NoMethodError) { storage.does_not_exist() }

    assert_equal(storage.public_send(:default_mount_by), Storage::MountByType_UUID)
    assert_raises(ArgumentError) { storage.public_send(:default_mount_by, "extra parameter") }
    assert_raises(NoMethodError) { storage.public_send(:does_not_exist) }

  end

end

File: bindings/python/testsuite/function-call.py
#!/usr/bin/python

import unittest
import storage


class TestPolymorphism(unittest.TestCase):

    def test_polymorphism(self):

        environment = storage.Environment(True, storage.ProbeMode_NONE, storage.TargetMode_DIRECT)
        s = storage.Storage(environment)

        self.assertEqual(s.get_default_mount_by(), storage.MountByType_UUID)
        self.assertRaises(TypeError, lambda: s.get_default_mount_by("extra-parameter"))
        self.assertRaises(AttributeError, lambda: s.does_not_exist())

        self.assertEqual(getattr(s, "get_default_mount_by")(), storage.MountByType_UUID)
        self.assertRaises(TypeError, lambda: getattr(s, "get_default_mount_by")("extra-parameter"))
        self.assertRaises(AttributeError, lambda: getattr(s, "does_not_exist"))

if __name__ == '__main__':
    unittest.main()
